### PR TITLE
Add Barotropic2D EOS

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

Will be used in Newtonian Euler so that the EOS can be runtime and is always 2d.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/PointwiseFunctions/Hydro/EquationsOfState/Barotropic2D.hpp [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+namespace EquationsOfState {
+/*!
+ * \ingroup EquationsOfStateGroup
+ * \brief A 2D equation of state representing a barotropic fluid.
+ *
+ *
+ * The equation of state takes the form
+ *
+ * \f[
+ * p = p (T, rho, Y_e) = p(0, rho, Y_e= Y_{e, \beta})
```
Comment (kidder) [inline src/PointwiseFunctions/Hydro/EquationsOfState/Barotropic2D.hpp]: fix the equation (rho -> \rho)

File: src/PointwiseFunctions/Hydro/EquationsOfState/Barotropic2D.hpp [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+ * \f]
+ *
+ * where \f$\rho\f$ is the rest mass density, \f$T\f$  the
+ * temperatur , and \f$Y_e\f$ is the electron fraction are not
```
Comment (kidder) [inline src/PointwiseFunctions/Hydro/EquationsOfState/Barotropic2D.hpp]: temperature

Comment (kidder) [reply src/PointwiseFunctions/Hydro/EquationsOfState/Barotropic2D.hpp]: electron fraction.  The temperature and electron fraction are not used, so evaluating this EoS at any arbitrary temperature or electron fraction is...

File: src/PointwiseFunctions/Hydro/EquationsOfState/Barotropic2D.hpp [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+ * used, and therefore this evaluating this EoS at any arbtirary
+ * temeperature or electron fraction is equivalent to evaluating it at
+ * temperature 0 and in beta equalibrium
```
Comment (kidder) [inline src/PointwiseFunctions/Hydro/EquationsOfState/Barotropic2D.hpp]: zero temperature

File: src/PointwiseFunctions/Hydro/EquationsOfState/Barotropic2D.hpp [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+  }
+  static constexpr Options::String help = {
+      "A 2D EoS which is independent of electron fraction and temperature. "
+      "Contains an underlying 1D EoS which is dependent only "
+      "on rest mass density."};
+  struct UnderlyingEos {
+    using type = ColdEos;
+    static std::string name() { return pretty_type::short_name<ColdEos>(); }
+    static constexpr Options::String help{
+        "The underlying Eos which is being represented as a "
```
Comment (kidder) [inline src/PointwiseFunctions/Hydro/EquationsOfState/Barotropic2D.hpp]: Eos

File: src/PointwiseFunctions/Hydro/EquationsOfState/Barotropic2D.hpp [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+        "2D Eos.  Must be a 1D EoS"};
```
Comment (kidder) [inline src/PointwiseFunctions/Hydro/EquationsOfState/Barotropic2D.hpp]: EoS

File: src/PointwiseFunctions/Hydro/EquationsOfState/Barotropic2D.hpp [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+  };
+
+  using options = tmpl::list<UnderlyingEos>;
+
+  Barotropic2D() = default;
+  Barotropic2D(const Barotropic2D&) = default;
+  Barotropic2D& operator=(const Barotropic2D&) = default;
+  Barotropic2D(Barotropic2D&&) = default;
+  Barotropic2D& operator=(Barotropic2D&&) = default;
+  ~Barotropic2D() override = default;
```
Comment (kidder) [inline src/PointwiseFunctions/Hydro/EquationsOfState/Barotropic2D.hpp:71]: clang-tidy wants this to be virtual

File: tests/Unit/PointwiseFunctions/Hydro/EquationsOfState/Test_PiecewisePolytropicFluid.cpp [hunk 7]
```diff
@@ -342,6 +343,9 @@ SPECTRE_TEST_CASE(
   // Check the correct 3D EoS is got
   CHECK(*eos.promote_to_3d_eos() ==
         EoS::Barotropic3D<EoS::PiecewisePolytropicFluid<true>>(eos));
+  // Check the correct 2D EoS is got
```
Comment (kidder) [inline tests/Unit/PointwiseFunctions/Hydro/EquationsOfState/Test_PiecewisePolytropicFluid.cpp]: is obtained

**Review by kidder — CHANGES_REQUESTED** [review]
you can squash the changes

File: src/PointwiseFunctions/Hydro/EquationsOfState/Barotropic2D.hpp (hunk 6 above)
Comment (nilsdeppe) [reply src/PointwiseFunctions/Hydro/EquationsOfState/Barotropic2D.hpp:71]: I think this is a clang-tidy bug. `override` implies virtual

**Comment by nilsdeppe** [comment]
I addressed the comments (hopefully fully :D ) and just squashed directly. I also added a commit that makes the same fixes to the Barotropic3D eos. Thanks for the review!

**Review by kidder — APPROVED** [review]

**Pull request merged**